### DISCUSSION TITLE
How to debug analytics locally

### DISCUSSION THREAD

**Discussion by hayata-suenaga** [invertase/react-native-firebase#6497] [Q&A]
I have a managed Expo project that has `react-native-firebase/analytics`. I want to check if the events are logged correctly locally because these logs take time to show up on google/firebase analytics console.

I'm mainly developing on an iOS simulator. Is there any resource I can refer to figure out how to debug analytics events locally?

**Comment by mikehardy** [comment]
https://search.brave.com/search?q=firebase+analytics+debug+mode

https://firebase.google.com/docs/analytics/debugview

https://firebase.google.com/docs/analytics/debugview#ios+

Expo may be hiding all the complexity of a native mobile app, so I'm not sure how you go about this in an Expo managed project, but you'll need to find your .xcworkspace folder and edit the current build scheme to add that build flag then rebuild.

You may find it's easier debug this on android, where you do it with an adb command directly on the emulator

**Comment by bradjones1** [comment]
Added a feature request for an Expo config plugin on Canny: https://invertase.canny.io/react-native-firebase/p/expo-plugin-for-analyticsdebugview-configuration

**Reply by mikehardy** [reply]
Just a note that as far as I know, no one else at Invertase uses Expo, and I do not either. So from that perspective a feature request is interesting, but the expectation I would have is: if you (or someone else) wants an Expo config plugin, they need to post a PR with an Expo config plugin. I'm happy to host them and carry them forward over time (a large / has costs associated commitment!) but as a non-Expo user I am not able to author them.

**Comment by bradjones1** [comment]
Update: Using `adb` when connected to a computer with USB debugging enabled works for both production and Expo development builds. This is still pertinent for iOS, which requires a build flag, but this is pretty straightforward/works out of the box for Android+Expo.

**Comment by vanGalilea** [comment]
@mikehardy It seems that supporting debug mode in an Expo app has become even more important as Expo's adoption has grown. React Native's getting started guide now states: "If you’re building a new app with React Native, we recommend using a Framework."

While you can use React Native without a framework, most developers benefit from using one like Expo, which offers features such as file-based routing, high-quality universal libraries, and the ability to write plugins that modify native code without managing native files. You can check it out here: [React Native Environment Setup](https://reactnative.dev/docs/environment-setup).

**Reply by mikehardy** [reply]
Hi there @vanGalilea - it seems that an interested Expo user will need to write an Expo config plugin to add the build flag? Or perhaps there is an expo-specific way to add build flags already, at which point someone will need to document it

**Comment by quememo** [comment]
Any update on this? I don't know how to apply the flag used for iOS on an Expo project (https://firebase.google.com/docs/analytics/debugview?hl=es-419#ios+)

**Reply by mikehardy** [reply]
In an open source repository, the updates you see are the updates that exist.

>  it seems that an interested Expo user will need to write an Expo config plugin to add the build flag? Or perhaps there is an expo-specific way to add build flags already, at which point someone will need to document it
